### PR TITLE
Use literal `None` in session cookie samesite setting

### PR DESCRIPTION
**Pull request by stsewd**
Ref https://read-the-docs.sentry.io/issues/5420149653/?project=148442&query=is%3Aunresolved+issue.priority%3A%5Bhigh%2C+medium%5D&referrer=issue-stream&statsPeriod=7d&stream_index=5

### CODE REVIEW

File: readthedocs/core/middleware.py [hunk 1]
```diff
@@ -139,7 +139,7 @@ def process_response(self, request, response):
                         )
 
                         # NOTE: This was added to support the fallback cookie
-                        if not settings.SESSION_COOKIE_SAMESITE:
+                        if settings.SESSION_COOKIE_SAMESITE != "None":
```
Comment (humitos) [inline readthedocs/core/middleware.py:142]: Does this works when this setting is set to `False`? It seems it's a valid value: https://docs.djangoproject.com/en/4.2/ref/settings/#std-setting-SESSION_COOKIE_SAMESITE

File: readthedocs/core/middleware.py [hunk 2]
```diff
@@ -57,15 +57,15 @@ def process_request(self, request):
                 request.session = SessionBase()  # create an empty session
                 return
 
-        if settings.SESSION_COOKIE_SAMESITE:
-            super().process_request(request)
-        else:
+        if settings.SESSION_COOKIE_SAMESITE == "None":
```
Comment (humitos) [inline readthedocs/core/middleware.py:60]: It seems the logic is inverted here now. Before, when the setting was `None` we were calling `super()` but now we are running the code inside the if.

```
>>> if None:
...   print("None")
... 
>>> 
```

**Review by humitos — COMMENTED** [review]
I made some comments on the logic because it seems it's wrong to me.

File: readthedocs/core/middleware.py (hunk 1 above)
Comment (stsewd) [reply readthedocs/core/middleware.py:142]: We need to send `None`, False will use the browsers default. None will send the cookie on cross-site requests, we need that.

File: readthedocs/core/middleware.py (hunk 2 above)
Comment (stsewd) [reply readthedocs/core/middleware.py:60]: The else statement was moved to the if, so the logic remains the same.

**Review by humitos — APPROVED** [review]

**Pull request merged**